tests/backup/backup_test.go

```diff
@@ -2722,6 +2722,292 @@ var _ = Describe("{BackupAlternatingBetweenLockedAndUnlockedBuckets}", func() {
+			// Get user context
+			ctxNonAdmin, err := backup.GetNonAdminCtx(userName, password)
+			log.FailOnError(err, "Fetching %s ctx", userName)
+
+			// Register Source and Destination cluster
+			log.InfoD("Registering Source and Destination clusters from user context")
+			err = CreateSourceAndDestClusters(orgID, "", "", ctxNonAdmin)
+			dash.VerifyFatal(err, nil, "Creating source and destination cluster")
+
+			for _, accessLevel := range accesses {
+
+				log.InfoD("Current Access is  %v", accessLevel)
+
+				log.InfoD("share all backups with %v in source cluster with a user", accessLevel)
+				err := ClusterUpdateBackupShare(SourceClusterName, nil, []string{userName}, accessLevel, true, ctx)
+				log.FailOnError(err, "Failed sharing all backups for cluster [%s]", SourceClusterName)
+
+				userBackups, err := GetAllBackupsForUser(userName, password)
+				log.InfoD("len(userBackups %v", len(userBackups))
+				log.FailOnError(err, "Fail on Fetching backups for %s", userName)
+
+				restoreName := fmt.Sprintf("%s-%v", RestoreNamePrefix, time.Now().Unix())
+				ValidateSharedBackupWithUsers(userName, ctxNonAdmin, accessLevel, userBackups[len(userBackups)-1], restoreName)
+				if accessLevel != ViewOnlyAccess {
+					restoreNames = append(restoreNames, restoreName)
+				}
+				log.InfoD("RestoreNames - %v", restoreNames)
+				if accessLevel == FullAccess {
+					log.InfoD("The fullaccess exit begins ")
+					break
+				}
+				//wait for 15 mins
+				log.InfoD("wait for 15 mins")
+				time.Sleep(15 * time.Minute)
+				fetchedUserBackups, err := GetAllBackupsForUser(userName, password)
+				log.FailOnError(err, "Fail on Fetching backups for %s", userName)
+				log.InfoD("All the backups for user %s - %v", userName, fetchedUserBackups)
+
+				recentBackupName := fetchedUserBackups[len(fetchedUserBackups)-1]
+				log.InfoD("recent backup - %v ", recentBackupName)
+
+				//Check if Schedule Backup came up or not
+				dash.VerifyFatal(len(fetchedUserBackups), (len(userBackups) + 1), "Verifying the new schedule backup is up or not")
+
+				//Now get the status of new backup -
+				backupSuccessCheck := func() (interface{}, bool, error) {
+					backupDriver := Inst().Backup
+					bkpUid, err := backupDriver.GetBackupUID(ctx, recentBackupName, orgID)
+					log.FailOnError(err, "Failed while trying to get backup UID for - %s", recentBackupName)
+					backupInspectRequest := &api.BackupInspectRequest{
+						Name:  recentBackupName,
+						Uid:   bkpUid,
+						OrgId: orgID,
+					}
+					resp, err := backupDriver.InspectBackup(ctx, backupInspectRequest)
+					log.FailOnError(err, "Inspecting the backup taken with request:\n%v", backupInspectRequest)
+					actual := resp.GetBackup().GetStatus().Status
+					expected := api.BackupInfo_StatusInfo_Success
+					if actual != expected {
+						return "", true, fmt.Errorf("backup status for [%s] expected was [%s] but got [%s]", recentBackupName, expected, actual)
+					}
+					return "", false, nil
+				}
```

Comment:
return for all steps in this go func() and then check the return `err` content after the `DoRetryWithTimeout`. Stay consistent here, don't do half `log.FailOnError()` and half `return "", false/true, nil/err`